shell/navi.plugin.ps1

```diff
@@ -1,60 +1,35 @@
-    ### then quit with a successful exit code.
-    ### We're not supposed to update the shortcut binding.
-    if ($Update){
-        Write-Debug "Updating Navi-Widget PowerShell module"
-        Copy-Item -Path .\navi.plugin\ -Destination $PwshModulePath -Force -Recurse
-        exit 0
-    }
-
-    ### If we're not updating, check if we don't have the module already installed
-    if (-Not (Test-Path -Path $PwshModulePath\$NAVI_PLUGIN)) {
-        Write-Debug "Copying Item to the path"
-        Copy-Item -Path .\navi.plugin\ -Destination $PwshModulePath -Recurse
-    } else {
-        Write-Error "Navi-Widget is already installed for PowerShell!"
-        exit 1
+$null = New-Module {
+    ### Initial code from @lurebat (https://github.com/lurebat/)
+    ### See #570 (https://github.com/denisidoro/navi/issues/570) for its original contribution
+    function Invoke-NaviWidget {
+        $ast = $tokens = $errors = $cursor = $null
+        [Microsoft.PowerShell.PSConsoleReadLine]::GetBufferState([ref] $ast, [ref] $tokens, [ref] $errors, [ref] $cursor)
+
+        $line = $ast.ToString().Trim()
+        $output = $null
+
+        if ([String]::IsNullOrEmpty($line)) {
+            $output = navi --print
+        }
+        else {
+            $best_match = (navi --print --best-match --query $line | Out-String).Trim()
+            if ([String]::IsNullOrEmpty($best_match)) {
+                $output = (navi --print --query "$line" | Out-String).Trim()
+            }
+            else {
+                $output = $best_match
+            }
+        }
+
+        [Microsoft.PowerShell.PSConsoleReadLine]::RevertLine()
+
+        ### Handling the case when the user escapes without selecting any entry
+        if (-Not([String]::IsNullOrEmpty($output))) {
+            [Microsoft.PowerShell.PSConsoleReadLine]::Insert([String]$output)
+        }
     }
 
-    Write-Debug "Registering the navi shortcut inside the current shell session"
     Set-PSReadlineKeyHandler -BriefDescription "A keybinding to open Navi Widget" -Chord Ctrl+g -ScriptBlock { Invoke-NaviWidget }
-
-    Write-Debug "Appending the navi shortcut inside the current user's profile"
-    ### Adding a new line
-    Write-Output "Import-Module navi.plugin" >> $PROFILE
-    Write-Output "" >> $PROFILE
-    Write-Output 'Set-PSReadlineKeyHandler -BriefDescription "A keybinding to open Navi Widget" -Chord Ctrl+g -ScriptBlock { Invoke-NaviWidget }' >> $PROFILE
-
-
-    Write-Output "Navi plugin has been installed!"
-    exit 0 ### Succesful installation
-}
-
-if ($Verbose) {
-    ### Enabling verbose/debug output at the start of the script
-    $DebugPreference = 'Continue'
-    InstallNaviWidgetModule -Update $Update
-    ### Disabling verbose/debug output at the end of the script
-    ### in order to not modify the current user's shell environment
-    $DebugPreference = 'SilentlyContinue'
-} else {
-    InstallNaviWidgetModule -Update $Update
 }
```

Comment:
By default, functions in a module are exported.
It is better to prevent functions from being exported.

```suggestion

    Export-ModuleMember -Function @()
}
```